mesheryctl/internal/cli/root/system/start.go

```diff
@@ -304,7 +318,60 @@ func start() error {
 
 		log.Info("...Meshery deployed on Kubernetes.")
 
-	// switch to default case if the platform specified is not supported
+		clientset := client.KubeClient
+
+		var opts meshkitkube.ServiceOptions
+		opts.Name = "meshery"
+		opts.Namespace = utils.MesheryNamespace
+		opts.APIServerURL = client.RestConfig.Host
+
+		var endpoint *meshkitutils.Endpoint
+
+		deadline := time.Now().Add(3 * time.Second)
+
+		//polling for endpoint to be available within the three second deadline
+		for !(time.Now().After(deadline)) {
+			endpoint, err = meshkitkube.GetServiceEndpoint(context.TODO(), clientset, &opts)
+			if err == nil {
+				break
+			} else {
+				time.Sleep(1 * time.Second)
+			}
+		}
+
+		if err != nil {
+			return err
+		}
+
+		currCtx.Endpoint = "http://" + endpoint.External.Address + ":" + strconv.Itoa(int(endpoint.External.Port))
```

Comment:
The assumption that SSL (https) is not being used is dangerous. Can this be centralized as a property?

Reply:
@leecalcote I tried removing it and the default is `https`, which resulted in the error: `Kubernetes unreachable`. 
I was thinking this could be centralised in `helpers.go`?
Thanks!

Reply:
@navendu-pottekkat @kumarabd comments on the central location? // @s1ntaxe770r

Reply:
The central location is in the utils package...

Reply:
Yes @navendu-pottekkat , `helpers.go` is part of the `utils` package.
I have pushed a commit centralising it in `utils` package.

Reply:
Yes, I missed your comment